src/Common/tests/System/Net/VirtualNetwork/VirtualNetwork.cs

```diff
@@ -36,7 +36,24 @@ public void ReadFrame(bool server, out byte[] buffer)
 
             semaphore.Wait(WaitForReadDataTimeoutMilliseconds);
 
-            bool dequeueSucceeded = packetQueue.TryDequeue(out buffer);
+            bool dequeueSucceeded = false;
+            int remainingRetries = 3;
+            int backOffDelayMilliseconds = 10;
+
+            do
+            {
+                dequeueSucceeded = packetQueue.TryDequeue(out buffer);
```

Comment:
I'm not following.

The reason TryDequeue might return false is if there's nothing in the collection.  And the only reason that might happen is if the semaphore.Wait(timeout) returned false due to hitting the timeout (if it returned true, that meant that there was count in the semaphore that could be taken, and that would only happen if there was a corresponding item to be taken from the queue).

So I'm suggesting that instead of wrapping TryDequeue in a loop, you wrap the Wait in a loop.  If the Wait returns true, TryDequeue should never return false, and if the Wait returns false, there's no reason to even try calling TryDequeue.